pkg/util/kernel/find_headers.go

```diff
@@ -110,12 +135,32 @@ func validateHeaderDirs(hv Version, dirs []string) error {
 		if dirv != hv {
 			return fmt.Errorf("header version %s does not match host version %s", dirv, hv)
 		}
+
 		// as long as one directory passes, validate the entire set
-		log.Debugf("found valid kernel headers at %s", d)
-		return nil
+		versionMatches = true
 	}
 
-	return fmt.Errorf("no valid kernel headers found")
+	if !versionMatches {
+		return fmt.Errorf("no valid kernel headers found")
+	}
+
+	if !linuxTypesHFound {
+		return fmt.Errorf("correctly versioned kernel headers found, but linux/types.h missing")
+	}
+
+	log.Debugf("valid kernel headers found in %v", dirs)
+	return nil
+}
+
+func containsLinuxTypesHFile(dir string) bool {
+	path := filepath.Join(dir, "include/linux/types.h")
+	f, err := os.Open(path)
+	if os.IsNotExist(err) {
```

Comment:
This is how the docs recommend to check for this now. You will need to import `io/fs` too.
```suggestion
	if errors.Is(err, fs.ErrNotExist) {
```